### DISCUSSION TITLE
Need help with blocking an element depending on nested content

### DISCUSSION THREAD

**Discussion by Feathered-Serpent** [uBlockOrigin/uBlock-issues#2625] [Q&A] [answered]
Hey everyone,

I hope I could get some help here. German newspage morgenpost.de has payed content, that I just don't want to see listed at all.

with my little knowledge, I found out, that with the following filter, I can at least remove the article picture from the main page (that with the green M+ in the picture):
`www.morgenpost.de##.ispayedcontent`

Is it possible though to remove the full <article - </article block when the ispayedcontent thing is included within somewhere?

**Comment by peace2000** [comment]
How about:

`morgenpost.de##article.teaser[data-article-id]:has(.ispayedcontent)`

**Comment by mapx-** [comment, marked as answer]
add

```
morgenpost.de##article:has(.ispayedcontent)
morgenpost.de##.content-quad__column:has(.ispayedcontent)
morgenpost.de###plus-banner--sticky
```

**Reply by peace2000** [reply]
`morgenpost.de##.content-quad__column:has(.ispayedcontent)`

Seems to hide regional categories at the bottom page and a free article.

**Reply by Feathered-Serpent** [reply]
I compared the website with the three filters active: https://i.imgur.com/98Sqk6b.png
I couldn't see which free article was hiding. Or was it not within the regional categories?

**Reply by peace2000** [reply]
In your picture e.g. a regional category "Friedrichshain-Kreuzberg" is blocked because there's a paid article below it. I think that the paid article should be blocked of course but not the category, because you should be able to access categories.


>I couldn't see which free article was hiding. Or was it not within the regional categories?

I said a little misleadingly. It blocks a whole category, if there's at least one paid article within it:

https://user-images.githubusercontent.com/17256841/235302322-bb48b18e-f892-4366-a5c9-d7784341b785.png

In this case, category of Berlin news because there's one paid article within it.

**Reply by Feathered-Serpent** [reply]
Ah, now I understand! As one can still click on the flags of the regions atop of the section, I think for me that should be ok.

Thanks for both of your help! :D
